sql/analyzer/resolve_having.go

```diff
@@ -104,37 +104,46 @@ func projectOriginalAggregation(having *plan.Having, schema sql.Schema) *plan.Pr
 
 var errHavingChildMissingRef = errors.NewKind("cannot find column %s referenced in HAVING clause in either GROUP BY or its child")
 
-func pullMissingColumnsUp(having *plan.Having, missingCols []string) (*plan.Having, error) {
-	groupBy, err := findGroupBy(having)
-	if err != nil {
-		return nil, err
-	}
-
-	schema := groupBy.Child.Schema()
+// pullMissingColumnsUp will attempt to find given missing columns. It will traverse on plan.Having node and scan
+// its children's schema to find the missing columns. The columns that are found will be added in Projections of
+// underlying plan.Project node and SelectExprs of underlying plan.GroupBy node.
```

Comment:
Seeing these function docs makes me so happy! Thank you for taking the time to add them! 😁